pkg/timeutil/timespan.go

```diff
@@ -0,0 +1,186 @@
+package timeutil
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	badDay = time.Weekday(-1)
+
+	shortDayNames = map[string]time.Weekday{
+		"Sun": time.Sunday,
+		"Mon": time.Monday,
+		"Tue": time.Tuesday,
+		"Wed": time.Wednesday,
+		"Thu": time.Thursday,
+		"Fri": time.Friday,
+		"Sat": time.Saturday,
+	}
+)
+
+func d2s(d time.Weekday) string {
+	for k, v := range shortDayNames {
+		if d == v {
+			return k
+		}
+	}
+
+	return ""
+}
+
+type dayTime struct {
+	day     time.Weekday // day of week, -1 if unset
+	minutes int          // minutes into day
+}
+
+func parseClock(s string) (hh, mm int, err error) {
+	spl := strings.Split(s, ":")
+	if len(spl) != 2 {
+		err = errors.New("malformed HH:MM specification")
+		return
+	}
+
+	hh, err = strconv.Atoi(spl[0])
+	if err != nil {
+		err = fmt.Errorf("malformed hour %q: %s", spl[0], err)
+		return
+	}
+
+	mm, err = strconv.Atoi(spl[1])
+	if err != nil {
+		err = fmt.Errorf("malformed minute %q: %s", spl[1], err)
+		return
+	}
+
+	return
+}
+
+func parsedaytime(s string) (dayTime, error) {
+	dt := dayTime{badDay, 0}
+	spl := strings.Fields(s)
+
+	switch len(spl) {
+	default:
+		fallthrough
+	case 0:
+		return dt, errors.New("empty time specification")
+	case 2:
+		// 2 fields, day hh:mm
+		for s, wd := range shortDayNames {
```

Comment:
populate the map with lowers, and just do a lookup into the map using the lowered form of spl[0], searching a map linearly by iterating over its keys defeats the purpose of using a map :).

That is, assuming we can't just do this conversion of weeday string to time.Weekday using the time package (I glanced, didn't see anything obvious, sigh)